### ISSUE TITLE
are there Typescript types?

### ISSUE THREAD

**Issue by swiknaba** [shrpne/vue-inline-svg#28]
thanks for the library!

Are there any plans on adding type definitions? Possibly, some progress is already made?

Need help with that? 🙂

**Comment by shrpne** [comment]
Hi! I'm not experienced with Typescript, so I'm not sure that I'm able to maintain types.

If you want to add types and maintain them, I don't mind. Help is always welcome :)

**Comment by eddie-englund** [comment]
This would be awesome!

Because vue3 with Ts cast's errors like these:

```bash
Could not find a declaration file for module 'vue-inline-svg'. 'C:/Users/Titus/code/portfolio/app/node_modules/vue-inline-svg/dist/vue-inline-svg.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/vue-inline-svg` if it exists or add a new declaration (.d.ts) file containing `declare module 'vue-inline-svg';`
```

Also, you can use https://github.com/Microsoft/dts-gen to generate the d.ts

**Comment by eddie-englund** [comment]
Just made a PR, for such a generated file :)

Whenever you make changes to the props, just install this NPM package globally, then run the below command in the code root directory:

```bash
dts-gen -m vue-inline-svg
```

**Comment by eddie-englund** [comment]
Actually, I tested it locally and had some issues, I'll work more on this tomorrow and hopefully I'll get a working types file for this project 👍

**Comment by shrpne** [comment]
Please review #32 
It is my attempt to add types, If it's ok, I will merge it

**Comment by vincerubinetti** [comment]
Update on this? Would be nice to get this integrated :)

**Comment by vincerubinetti** [comment]
For what it's worth, I tested out the PR and it seems to work. Though, I am not very experienced with typescript, and I also had to change `VueConstructor` to `VueElementConstructor` because I am using Vue 3 (no idea if that is actually the proper type/class, but it made the typescript errors go away 🤷 ).

**Comment by shrpne** [comment]
@vincerubinetti thank you, i will try to handle it soon

**Comment by shrpne** [comment]
Types added in aeacd342ba7f616e783247875c32b59183e2018a for vue3, published under [v3.1.0](https://github.com/shrpne/vue-inline-svg/releases/tag/v3.1.0)

And 32b306ae49adb9e3483a4425ccc3362ce084d7f2 for vue2, published under [v2.1.0](https://github.com/shrpne/vue-inline-svg/releases/tag/v2.1.0)
